src/components/SecretsTable/SecretsTable.js

```diff
@@ -28,124 +39,160 @@ const {
+                        <TableHeader
+                          {...getHeaderProps({ header })}
                           className="cellText"
+                          key={header.key}
                         >
-                          {cell.value}
-                        </TableCell>
-                      ))}
-                      <TableCell
-                        key={lastCell.id}
-                        id={lastCell.id}
-                        className="cellIcon"
-                        onClick={() => {
-                          handleDelete({
-                            name: secrets[index].name,
-                            namespace: secrets[index].namespace
-                          });
-                        }}
-                        data-testid="deleteIcon"
+                          {header.header}
+                        </TableHeader>
+                      );
+                    })}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {rows.map(row => {
+                    return (
+                      <TableRow
+                        {...getRowProps({ row })}
+                        key={row.id}
+                        className="row"
                       >
-                        {lastCell.value}
-                      </TableCell>
-                    </TableRow>
-                  );
-                })}
-              </TableBody>
-            ) : (
-              <TableBody>
-                <TableRow className="row">
-                  {headers.splice(0, headers.length - 1).map(header => (
-                    <TableCell
-                      key={`${header.key}-empty`}
-                      id={`${header.key}-empty`}
-                      className="cellTextNone"
-                    >
-                      -
-                    </TableCell>
-                  ))}
-                  <TableCell id="icon-empty" className="cellIconNone">
-                    -
-                  </TableCell>
-                </TableRow>
-              </TableBody>
+                        <TableSelectRow {...getSelectionProps({ row })} />
+                        {row.cells.map(cell => (
+                          <TableCell
+                            key={cell.id}
+                            id={cell.id}
+                            className="cellText"
+                          >
+                            {cell.value}
+                          </TableCell>
+                        ))}
+                      </TableRow>
+                    );
+                  })}
+                </TableBody>
+              </Table>
             )}
-          </Table>
-        </TableContainer>
+          </TableContainer>
+        )}
+      />
+      {secrets.length === 0 && selectedNamespace === ALL_NAMESPACES && (
+        <div className="noSecrets">
+          <p>
+            No secrets created under any namespace, click the button above to
```

Comment:
avoid references to layout / position (above, left, etc.) in user-facing text as this is not accessible to users of screen readers, and also tightly couples the text to the design

Reply:
I see, is there some other way to let the user know that there are no secrets in the given namespace or all-namepaces? 

this is what i was trying to achieve:
<img width="1402" alt="Screen Shot 2019-07-15 at 2 54 06 PM" src="https://user-images.githubusercontent.com/49996607/61240957-71149380-a710-11e9-8783-dbbd9ee28c33.png">

Reply:
You can still tell the user that there are no secrets, the only part of the string that's problematic is the use of 'above'. You could replace it with 'click the Create Secret button to add one' or similar.

Reply:
ohh i understand now. awesome!